dell/xps/15-7590/default.nix

```diff
@@ -0,0 +1,39 @@
+{ lib, ... }:
+
+{
+  imports = [
+    ../../../common/cpu/intel
+    ../../../common/pc/laptop
+    ../../../common/pc/ssd
+  ];
+
+  # Set to true for just the first run, then disable it.
+  # boot.loader.efi.canTouchEfiVariables = lib.mkDefault true;
+
+  # Earlier font-size setup
+  console.earlySetup = true;
+
+  # Prevent small EFI partiion from filling up
+  boot.loader.grub.configurationLimit = 10;
+
+  # The 48.ucode causes the Killer wifi card to crash.
+  # The iwlfwifi-cc-a0-46.ucode works perfectly
+  nixpkgs.pkgs = import <nixpkgs> {
```

Comment:
Wouldn't this prevent the user from specifying their own `nixpkgs.config` options or doing cross platform builds of the configuration? I guess it may also interfere if you are deploying this configuration using nixops.

Reply:
Please see: https://github.com/NixOS/nixos-hardware/pull/143